src/shared_modules/content_manager/tests/component/action_test.cpp

```diff
@@ -271,6 +225,41 @@ TEST_F(ActionTest, TestInstantiationAndRunActionOnDemand)
     EXPECT_TRUE(std::filesystem::exists(contentPath));
 
     EXPECT_TRUE(std::filesystem::exists(outputFolder));
+}
+
+/**
+ * @brief Tests the on-start execution of the action.
+ *
+ */
+TEST_F(ActionTest, ActionOnStartExecution)
+{
+    constexpr auto ACTION_INTERVAL {100};
+    constexpr auto WAIT_TIME {1};
+
+    const auto& topicName {m_parameters.at("topicName").get_ref<const std::string&>()};
+    const auto& outputFolder {m_parameters.at("configData").at("outputFolder").get_ref<const std::string&>()};
+    const auto& fileName {m_parameters.at("configData").at("contentFileName").get_ref<const std::string&>()};
+
+    // Make the interval big enough to be sure the action is not triggered a second time.
+    auto& interval {m_parameters.at("interval").get_ref<size_t&>()};
+    interval = ACTION_INTERVAL;
+
+    // Init action.
+    auto action {std::make_shared<Action>(m_spRouterProvider, topicName, m_parameters)};
+
+    // Check output folder existence.
+    EXPECT_TRUE(std::filesystem::exists(outputFolder));
+
+    // Start scheduling.
+    EXPECT_NO_THROW(action->startActionScheduler(interval));
+
+    // Wait just for a little time.
+    std::this_thread::sleep_for(std::chrono::seconds(WAIT_TIME));
+
+    // Stop scheduling.
+    EXPECT_NO_THROW(action->stopActionScheduler());
 
-    EXPECT_NO_THROW(routerProvider->stop());
+    // Check that the download has been correctly made.
+    const auto contentFilePath {outputFolder + "/" + CONTENTS_FOLDER + "/3-" + fileName};
```

Comment:
Question: Where does the /3- comes from?

Reply:
The server response is returning a `last_offset` equal to 3: 

https://github.com/wazuh/wazuh/blob/32dbbaa7cf3f5665e3d8d05c9dfe339eb292b00b/src/shared_modules/content_manager/tests/component/fakes/fakeServer.hpp#L78

Then, the downloader fetches from offset 0 to 3, storing the output content in a file with the prefix 3:
https://github.com/wazuh/wazuh/blob/32dbbaa7cf3f5665e3d8d05c9dfe339eb292b00b/src/shared_modules/content_manager/src/components/CtiApiDownloader.hpp#L51

Reply:
Can we add a comment? Or better create a const with the "/3", I've seen that is used in many places

Reply:
> Or better create a const with the "/3"

I see that as an improvement, agreed, although not from this PR's scope. If I get the OK from @Damian-Mangold, it's done.

Reply:
is not the scope of this PR. It is noted for future improvement.

Reply:
LGTM then!